fvm/state/accounts_status.go

```diff
@@ -1,51 +1,112 @@
 package state
 
 import (
+	"encoding/binary"
 	"encoding/hex"
 
+	"github.com/onflow/atree"
+
 	"github.com/onflow/flow-go/fvm/errors"
 )
 
-type AccountStatus uint8
+const (
+	flagSize            = 1
+	storageUsedSize     = 8
+	storageIndexSize    = 8
+	publicKeyCountsSize = 8
+
+	accountStatusSize = flagSize +
+		storageUsedSize +
+		storageIndexSize +
+		publicKeyCountsSize
+
+	flagIndex                 = 0
+	storageUsedStartIndex     = flagIndex + flagSize
+	storageIndexStartIndex    = storageUsedStartIndex + storageUsedSize
+	publicKeyCountsStartIndex = storageIndexStartIndex + storageIndexSize
+)
+
+// AccountStatus holds meta information about an account
+//
+// currently modelled as a byte array with on-demand encoding/decoding of sub arrays
+// the first byte captures flags (e.g. frozen)
+// the next 8 bytes (big-endian) captures storage used by an account
+// the next 8 bytes (big-endian) captures the storage index of an account
+// and the last 8 bytes (big-endian) captures the number of public keys stored on this account
+type AccountStatus [accountStatusSize]byte
 
 const (
-	maskExist  byte = 0b0000_0001
 	maskFrozen byte = 0b1000_0000
 )
 
-// NewAccountStatus sets exist flag and return an AccountStatus
-func NewAccountStatus() AccountStatus {
-	return AccountStatus(maskExist)
+// NewAccountStatus returns a new AccountStatus
+func NewAccountStatus(storageIndex atree.StorageIndex) *AccountStatus {
+	var as AccountStatus
+	as.SetStorageIndex(storageIndex)
+	return &as
```

Comment:
Not sure, but if we only use default storage index for new account status, maybe we can just hard code the default index inside `NewAccountStatus`, instead of passing it as parameter.

Reply:
I thought about it but felt that's an account-level responsibility and not something account status should know, though from a performance perspective that might save use some copy time.

Reply:
updated at 58e5e9b - you're right if we never call it with different input it can have a default value built in.